### PR TITLE
:package: improve docs

### PR DESCRIPTION
**Pull request by jvdd**
This PR aims at improving the documentation of this package, as we have had several issues about the lacking documentation (#99, #91, #102)
- [x] create FAQ
- [x] update `Readme.md`
- [ ] add `Contributing.md`
- [ ] add `Changelog.md`
- [x] add requirements.txt to example folder
- [x] improve dash integration docs
- [x] add very minimal dash integration example
- [x] update example notebooks
- [x] add more documentation to the C code
- [x] incorporate figure serialization into the docs

##Other stuff this PR does;
- [x] enable numerically unstable test (see #93)

### incorporate Plotly-resampler's LTTBc bindings

TODO:
- [x] test lttb_core_py
- [x] test the `EfficientLTTB` method when `LTTB_core_c` is not available
  **note**: this method is tested for equality with the lttbc method

### CODE REVIEW

**Comment by jonasvdd** [comment]
Ready for review @jvdd!

**Comment by SirVer** [comment]
I looked through the examples in this MR and ran them on my system and I think this is a huge improvement:

- `01_minimal_global.py` worked out of the box for me, has minimal dependencies and a lot of really useful comments explaining what is going on, why having a global is bad practice. It contains the gist of how the library is meant to be used. Really great!
- `02_minimal_cache.py` seems to build logically on 01, infact looking at the diff between the files nicely shows the differences needed for caching. however, when running it and zooming a bit I get an error ([traceback in this Gist](https://gist.github.com/722ddd00d7b9ccae0ad7a75180f4dd66)).  
- `03_minimal_cache_dynamic.py` is clear in its instructions, but not in its motivation. Why would I want to use a dcc.Store in the first place? When is this useful? Running it I get a different error when zooming the dynamically added graph, but similar looking error to above (`ValueError: Invalid property specified for object of type plotly.graph_objs.Scatter: 'data'`).


I did not look at the remaining examples, they seemed much more advanced to me.

Great job at improving the examples! I'll go ahead and try to implement this in my app now again.

**Comment by jonasvdd** [comment]
Hi @SirVer, 

I cannot seem to reproduce your experienced issues; could you share a pip freeze; so my python environment is an exact copy of yours?

**Comment by SirVer** [comment]
@jonasvdd Here is [a gist](https://gist.github.com/SirVer/d88b85a6438c68dbb4453bd6cbee1195) with the Pipfile & Pipfile.lock that made my environment.

I did the following:
1. cd to `/tmp/plotly-resampler/examples/dash_apps`, which had this branch checkout out.
2. `pipenv shell`
3. `pipenv install dash plotly_resampler`
4. `python 02_minimal_cache.py`

**Comment by jonasvdd** [comment]
Thanks! I was able to reproduce the error. 🥲 

I think that the serialization support was not yet integrated into `0.7.2.2`

If I execute `pipenv install plotly-resampler==0.8.0rc12` and then run the examples; they work flawlessly!

**Comment by jonasvdd** [comment]
@jvdd; ready for review!

File: plotly_resampler/aggregation/aggregators.py [hunk 1]
```diff
@@ -274,8 +282,15 @@ def __init__(self, interleave_gaps: bool = True, nan_position="end"):
         )
 
     def _aggregate(self, s: pd.Series, n_out: int) -> pd.Series:
-        if s.shape[0] > n_out * 2_000:
-            s = self.minmax._aggregate(s, n_out * 50)
+        size_threshold = 10_000_000
+        ratio_threshold = 100
+
+        # TODO -> test this with a move of the .so file
```
Comment (jvdd) [inline plotly_resampler/aggregation/aggregators.py:288]: What do you mean? @jonasvdd

File: plotly_resampler/figure_resampler/utils.py [hunk 2]
```diff
@@ -4,7 +4,10 @@
 import pandas as pd
 
 from plotly.basedatatypes import BaseFigure
-from plotly.basewidget import BaseFigureWidget
+try:
+    from plotly.basewidget import BaseFigureWidget
```
Comment (jvdd) [inline plotly_resampler/figure_resampler/utils.py:8]: Why does this have to be in try-except?

**Review by jvdd — COMMENTED** [review]
LGTM! Very nice work, quality of docs is improved significantly :rocket:

File: plotly_resampler/figure_resampler/utils.py (hunk 2 above)
Comment (jonasvdd) [reply plotly_resampler/figure_resampler/utils.py:8]: As a result we do not need Ipywidgets as a required dependency (which is useful when you want to just use it for a dash-app)

**Review by jonasvdd — APPROVED** [review]

**Pull request merged**